## This Script runs all the notebook files one by one.

In [2]:
import numpy as np
import os
from os.path import join as opj

In [3]:
# fc_file_list = None

By mistake I over wrote the functionalConnectivity and corrupted the FC maps. but datasink is still intact

In [4]:
# Paths
path_cwd = os.getcwd()
path_split_list = path_cwd.split('/')
s = path_split_list[0:-1]
s = opj('/',*s) # *s converts list to path, # also it is very important to add '/' in the begining so it is read as directory later
base_directory = opj(s,'result')

# data_path = "data/ABIDE2_NYU1-BIDS-Preprocessed-GSR/NYU1/"
data_path = "data/ABIDE-BIDS-Preprocessed-GSR/NYU/"
wf_name = 'globalSignalRemoval_ABIDE1_UCLA1_GSR_FC'
datasink_name = 'datasink_ABIDE1_UCLA1_GSR_FC'

# % run functional_connectivity.ipynb

In [9]:
%%time
# Paths


# fc_file_path = opj(base_directory,'functionalConnectivity_GSR0_fisher0','save_fc_file_list/fc_file_list.npy')
fc_file_path = opj(base_directory,wf_name,'save_fc_file_list/fc_file_list.npy')
# fc_file_path = opj(base_directory,wf_name,'save_fc_file_list/fc_file_comp_list.npy')
# save_destination = opj(base_directory,wf_name)
# if not os.path.exists(save_destination):
#     os.mkdir(save_destination)


apply_fisher = True
fc_file_list = (np.load(fc_file_path))

# ABIDE1 - NYU
autism_start_index = 0
autism_end_index = 78
td_start_index = 79
td_end_index = 157 #183

# ABIDE2 - NYU1
# autism_start_index = 0
# autism_end_index = 29 # 47
# td_start_index = 48
# td_end_index = 77

# ABIDE1 UCLA_1
autism_start_index = 0
autism_end_index = 48
td_start_index = 49
td_end_index = 81

# GU1-NYU1
# participant_file = 'GU1_NYU1participants.tsv'
# pid_dx = np.loadtxt(participant_file,skiprows=1)

# autism_subject_indexes = np.array(np.where(pid_dx[:,1]==1)).squeeze()[0:84]
# td_subject_indexes = np.array(np.where(pid_dx[:,1]==2)).squeeze()[0:84]

autism_subject_indexes = np.arange(autism_start_index, autism_end_index+1 )[0:33]
td_subject_indexes = np.arange(td_start_index, td_end_index + 1 )

%run hypothesis_test-fast.ipynb

[ '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051201/pearcoff/0051201_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051202/pearcoff/0051202_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051203/pearcoff/0051203_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051204/pearcoff/0051204_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051205/pearcoff/0051205_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051206/pearcoff/0051206_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051207/pearcoff/0051207_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051208/pearcoff/0051208_fc_map.npy'
 '/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/_subject_id_0051209/pearcoff/0051209_fc_

Done subject  16
Done subject  17
Done subject  18
Done subject  19
Done subject  20
Done subject  21
Done subject  22
Done subject  23
Done subject  24
Done subject  25
Done subject  26
Done subject  27
Done subject  28
Done subject  29
Done subject  30
Done subject  31
Done subject  32
CPU times: user 2min 36s, sys: 2min 49s, total: 5min 26s
Wall time: 7min 26s
CPU times: user 2min 37s, sys: 2min 49s, total: 5min 27s
Wall time: 7min 27s


In [15]:
len(autism_subject_indexes),len(td_subject_indexes)

(79, 79)

In [59]:
autism_subject_indexes,td_subject_indexes

(array([ 11,  13,  15,  17,  20,  23,  24,  27,  30,  36,  37,  38,  39,
         41,  43,  46,  47,  48,  49,  50,  54,  55,  58,  59,  61,  64,
         66,  67,  68,  69,  70,  74,  75,  76,  77,  78,  79,  80,  81,
         82,  83,  84,  90,  91,  92,  93,  96,  97,  98, 101, 102, 106,
        107, 108, 109, 110, 111, 112, 113, 114, 115, 116, 117, 118, 119,
        120, 121, 122, 123, 124, 125, 126, 127, 128, 129, 130, 131, 132,
        133, 134, 135, 136, 137, 138]),
 array([  0,   1,   2,   3,   4,   5,   6,   7,   8,   9,  10,  12,  14,
         16,  18,  19,  21,  22,  25,  26,  28,  29,  31,  32,  33,  34,
         35,  40,  42,  44,  45,  51,  52,  53,  56,  57,  60,  62,  63,
         65,  71,  72,  73,  85,  86,  87,  88,  89,  94,  95,  99, 100,
        103, 104, 105, 154, 155, 156, 157, 158, 159, 160, 161, 162, 163,
        164, 165, 166, 167, 168, 169, 170, 171, 172, 173, 174, 175, 176,
        177, 178, 179, 180, 181, 182]))

In [60]:
dx

array([[  2.87410000e+04,   2.00000000e+00],
       [  2.87420000e+04,   2.00000000e+00],
       [  2.87430000e+04,   2.00000000e+00],
       [  2.87440000e+04,   2.00000000e+00],
       [  2.87450000e+04,   2.00000000e+00],
       [  2.87460000e+04,   2.00000000e+00],
       [  2.87470000e+04,   2.00000000e+00],
       [  2.87480000e+04,   2.00000000e+00],
       [  2.87490000e+04,   2.00000000e+00],
       [  2.87500000e+04,   2.00000000e+00],
       [  2.87510000e+04,   2.00000000e+00],
       [  2.87520000e+04,   1.00000000e+00],
       [  2.87530000e+04,   2.00000000e+00],
       [  2.87540000e+04,   1.00000000e+00],
       [  2.87550000e+04,   2.00000000e+00],
       [  2.87560000e+04,   1.00000000e+00],
       [  2.87570000e+04,   2.00000000e+00],
       [  2.87580000e+04,   1.00000000e+00],
       [  2.87590000e+04,   2.00000000e+00],
       [  2.87600000e+04,   2.00000000e+00],
       [  2.87610000e+04,   1.00000000e+00],
       [  2.87620000e+04,   2.00000000e+00],
       [  

In [10]:
fc_file_list

array([ '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28741/pearcoff/28741_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28742/pearcoff/28742_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28743/pearcoff/28743_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28744/pearcoff/28744_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28745/pearcoff/28745_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28746/pearcoff/28746_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28747/pearcoff/28747_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28748/pearcoff/28748_fc_map.npy',
       '/home1/varunk/result/functionalConnectivity_GU1_NYU1_GSR/_subject_id_28749/pearcoff/28749_fc_map.npy',


In [10]:
%%time
# paths
Pval_file_path = opj(base_directory,wf_name,'hypothesis_testing/Pvals.npy')

%run fdr_correction.ipynb

Job Done for ROI- 36
Job Done for ROI- 27
Job Done for ROI- 0
Job Done for ROI- 9
Job Done for ROI- 54
Job Done for ROI- 18
Job Done for ROI- 45
Job Done for ROI- 63
Job Done for ROI- 37
Job Done for ROI- 55
Job Done for ROI- 1
Job Done for ROI- 46
Job Done for ROI- 64
Job Done for ROI- 10
Job Done for ROI- 19
Job Done for ROI- 28
Job Done for ROI- 38
Job Done for ROI- 56
Job Done for ROI- 2
Job Done for ROI- 20
Job Done for ROI- 47
Job Done for ROI- 29
Job Done for ROI- 65
Job Done for ROI- 11
Job Done for ROI- 39
Job Done for ROI- 3
Job Done for ROI- 57
Job Done for ROI- 48
Job Done for ROI- 12
Job Done for ROI- 30
Job Done for ROI- 66
Job Done for ROI- 21
Job Done for ROI- 40
Job Done for ROI- 4
Job Done for ROI- 13
Job Done for ROI- 58
Job Done for ROI- 49
Job Done for ROI- 31
Job Done for ROI- 22
Job Done for ROI- 67
Job Done for ROI- 41
Job Done for ROI- 5
Job Done for ROI- 50
Job Done for ROI- 14
Job Done for ROI- 59
Job Done for ROI- 32
Job Done for ROI- 68
Job Done for ROI- 23

In [12]:
Pval_file_path

'/home1/varunk/result/globalSignalRemoval_ABIDE1_UCLA1_GSR_FC/hypothesis_testing/Pvals.npy'

In [11]:
res =(np.unique(np.where(fdrcorrected_brain_4Dtensor_mask[:,:,:,:] == 1)[3],  return_counts= True))
count_roi = zip(list(res[0]), list(res[1]))
set(count_roi)

{(2, 1),
 (6, 8),
 (24, 6),
 (25, 48),
 (30, 1),
 (53, 1),
 (62, 2),
 (64, 1),
 (77, 5),
 (95, 2),
 (130, 1),
 (142, 9),
 (150, 2),
 (152, 1),
 (153, 2),
 (174, 23),
 (175, 2),
 (189, 1),
 (198, 1),
 (204, 9),
 (222, 2),
 (226, 2)}

In [13]:
res =(np.unique(np.where(fdrcorrected_brain_4Dtensor_mask[:,:,:,:] == 1)[3],  return_counts= True))
count_roi = zip(list(res[0]), list(res[1]))
set(count_roi)

{(2, 1),
 (6, 8),
 (24, 6),
 (25, 48),
 (30, 1),
 (53, 1),
 (62, 2),
 (64, 1),
 (77, 5),
 (95, 2),
 (130, 1),
 (142, 9),
 (150, 2),
 (152, 1),
 (153, 2),
 (174, 23),
 (175, 2),
 (189, 1),
 (198, 1),
 (204, 9),
 (222, 2),
 (226, 2)}

In [18]:
opj(base_directory,wf_name,'save_fc_file_list/fc_file_list.npy')

'/home1/varunk/result/functionalConnectivity_GSR1_fisher0/save_fc_file_list/fc_file_list.npy'

In [14]:
autism_subject_indexes

array([ 0,  1,  2,  3,  4,  5,  6,  7,  8,  9, 10, 11, 12, 13, 14, 15, 16,
       17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32])

In [15]:
%%time
# Paths
fc_file_path = opj(base_directory,wf_name,'save_fc_file_list/fc_file_list.npy')
fc_file_list = (np.load(fc_file_path))
Pvals_file_path = opj(base_directory,wf_name,'hypothesis_testing/Pvals.npy')
Qvals_file_path = opj(base_directory,wf_name,'fdr_correction/Qvals.npy')

# participant_file = 'GU1_NYU1participants.tsv'
# pid_dx = np.loadtxt(participant_file,skiprows=1)

# autism_subject_indexes = np.array(np.where(pid_dx[:,1]==1)).squeeze()[0:84]
# td_subject_indexes = np.array(np.where(pid_dx[:,1]==2)).squeeze()[0:84]



%run results.ipynb

Autistic Participant index: 0
Autistic Participant index: 1
Autistic Participant index: 2
Autistic Participant index: 3
Autistic Participant index: 4
Autistic Participant index: 5
Autistic Participant index: 6
Autistic Participant index: 7
Autistic Participant index: 8
Autistic Participant index: 9
Autistic Participant index: 10
Autistic Participant index: 11
Autistic Participant index: 12
Autistic Participant index: 13
Autistic Participant index: 14
Autistic Participant index: 15
Autistic Participant index: 16
Autistic Participant index: 17
Autistic Participant index: 18
Autistic Participant index: 19
Autistic Participant index: 20
Autistic Participant index: 21
Autistic Participant index: 22
Autistic Participant index: 23
Autistic Participant index: 24
Autistic Participant index: 25
Autistic Participant index: 26
Autistic Participant index: 27
Autistic Participant index: 28
Autistic Participant index: 29
Autistic Participant index: 30
Autistic Participant index: 31
Autistic Participa

Stored ROI file the shared arrays! 
Job Done for ROI- 130
Stored ROI file the shared arrays! 
Job Done for ROI- 139
Stored ROI file the shared arrays! 
Job Done for ROI- 77
Stored ROI file the shared arrays! 
Job Done for ROI- 86
Stored ROI file the shared arrays! 
Job Done for ROI- 113
Stored ROI file the shared arrays! 
Job Done for ROI- 95
Stored ROI file the shared arrays! 
Job Done for ROI- 104
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Job Done for ROI- 131
Job Done for ROI- 122
Stored ROI file the shared arrays! 
Job Done for ROI- 140
Stored ROI file the shared arrays! 
Job Done for ROI- 78
Stored ROI file the shared arrays! 
Job Done for ROI- 87
Stored ROI file the shared arrays! 
Job Done for ROI- 114
Stored ROI file the shared arrays! 
Job Done for ROI- 96
Stored ROI file the shared arrays! 
Job Done for ROI- 105
Stored ROI file the shared arrays! 
Job Done for ROI- 132
Stored ROI file the shared arrays! 
Job Done for ROI- 123
Stored ROI file the 

Job Done for ROI- 248
Stored ROI file the shared arrays! 
Job Done for ROI- 239
Stored ROI file the shared arrays! 
Job Done for ROI- 221
Stored ROI file the shared arrays! 
Job Done for ROI- 231
Stored ROI file the shared arrays! 
Job Done for ROI- 266
Stored ROI file the shared arrays! 
Job Done for ROI- 257
Stored ROI file the shared arrays! 
Job Done for ROI- 240
Stored ROI file the shared arrays! 
Job Done for ROI- 249
Stored ROI file the shared arrays! 
Job Done for ROI- 222
Stored ROI file the shared arrays! 
Job Done for ROI- 267
Stored ROI file the shared arrays! 
Job Done for ROI- 232
Stored ROI file the shared arrays! 
Job Done for ROI- 241
Stored ROI file the shared arrays! 
Job Done for ROI- 250
Stored ROI file the shared arrays! 
Job Done for ROI- 258
Stored ROI file the shared arrays! 
Job Done for ROI- 223
Stored ROI file the shared arrays! 
Job Done for ROI- 268
Stored ROI file the shared arrays! 
Job Done for ROI- 251
Stored ROI file the shared arrays! 
Job Done for R

Job Done for ROI- 78
Stored ROI file the shared arrays! 
Job Done for ROI- 87
Stored ROI file the shared arrays! 
Job Done for ROI- 96
Stored ROI file the shared arrays! 
Job Done for ROI- 105
Stored ROI file the shared arrays! 
Job Done for ROI- 141
Stored ROI file the shared arrays! 
Job Done for ROI- 123
Stored ROI file the shared arrays! 
Job Done for ROI- 114
Stored ROI file the shared arrays! 
Job Done for ROI- 132
Stored ROI file the shared arrays! 
Job Done for ROI- 79
Stored ROI file the shared arrays! 
Job Done for ROI- 88
Stored ROI file the shared arrays! 
Job Done for ROI- 97
Stored ROI file the shared arrays! 
Job Done for ROI- 106
Stored ROI file the shared arrays! 
Job Done for ROI- 124
Stored ROI file the shared arrays! 
Job Done for ROI- 142
Stored ROI file the shared arrays! 
Job Done for ROI- 115
Stored ROI file the shared arrays! 
Job Done for ROI- 133
Stored ROI file the shared arrays! 
Job Done for ROI- 80
Stored ROI file the shared arrays! 
Job Done for ROI- 89


Job Done for ROI- 223
Stored ROI file the shared arrays! 
Job Done for ROI- 250
Stored ROI file the shared arrays! 
Job Done for ROI- 268
Stored ROI file the shared arrays! 
Job Done for ROI- 232
Stored ROI file the shared arrays! 
Job Done for ROI- 241
Stored ROI file the shared arrays! 
Job Done for ROI- 259
Stored ROI file the shared arrays! 
Job Done for ROI- 269
Stored ROI file the shared arrays! 
Job Done for ROI- 251
Stored ROI file the shared arrays! 
Job Done for ROI- 224
Stored ROI file the shared arrays! 
Job Done for ROI- 242
Stored ROI file the shared arrays! 
Job Done for ROI- 233
Stored ROI file the shared arrays! 
Job Done for ROI- 260
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Job Done for ROI- 45
Job Done for ROI- 54
Job Done for ROI- 36
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Job Done for ROI- 0
Job D

Stored ROI file the shared arrays! 
Job Done for ROI- 115
Stored ROI file the shared arrays! 
Job Done for ROI- 133
Stored ROI file the shared arrays! 
Job Done for ROI- 124
Stored ROI file the shared arrays! 
Job Done for ROI- 142
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Job Done for ROI- 98
Job Done for ROI- 80
Stored ROI file the shared arrays! 
Job Done for ROI- 89
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Job Done for ROI- 107
Job Done for ROI- 116
Stored ROI file the shared arrays! 
Stored ROI file the shared arrays! 
Job Done for ROI- 125
Job Done for ROI- 134
Stored ROI file the shared arrays! 
Job Done for ROI- 143
Stored ROI file the shared arrays! 
Job Done for ROI- 144
Stored ROI file the shared arrays! 
Job Done for ROI- 153
Stored ROI file the shared arrays! 
Job Done for ROI- 162
Stored ROI file the shared arrays! 
Job Done for ROI- 171
Stored ROI file the shared arrays! 
Job Done for ROI- 180
Stored ROI file t

Job Done for ROI- 260
CPU times: user 54.2 s, sys: 1min 9s, total: 2min 3s
Wall time: 7min 24s
Stats computed for ROI:  0
Stats computed for ROI:  8
Stats computed for ROI:  1
Stats computed for ROI:  2
Stats computed for ROI:  3
Stats computed for ROI:  4
Stats computed for ROI:  5
Stats computed for ROI:  6
Stats computed for ROI:  7
Stats computed for ROI:  9
Stats computed for ROI:  17
Stats computed for ROI:  10
Stats computed for ROI:  11
Stats computed for ROI:  12
Stats computed for ROI:  13
Stats computed for ROI:  14
Stats computed for ROI:  15
Stats computed for ROI:  16
Stats computed for ROI:  18
Stats computed for ROI:  26
Stats computed for ROI:  19
Stats computed for ROI:  20
Stats computed for ROI:  21
Stats computed for ROI:  22
Stats computed for ROI:  23
Stats computed for ROI:  24
Stats computed for ROI:  25
Stats computed for ROI:  27
Stats computed for ROI:  35
Stats computed for ROI:  28
Stats computed for ROI:  29
Stats computed for ROI:  30
Stats computed for 

In [44]:
wf_name
# To Stop execution Raise error:
# raise Exception('Execution stops here!')

'functionalConnectivity_GSR0_fisher1'

(ERROR) Results:

ROIs along with their voxels significant with q < 0.05  
Options:   
Fisher's transform  = True  
Global Signal Removed = True  
{(ROI,Voxels)}  

{(3, 1),
 (5, 3),
 (18, 40),
 (20, 51),
 (21, 15),
 (24, 88),
 (30, 4),
 (37, 21),
 (38, 12),
 (40, 506),
 (41, 862),
 (44, 55),
 (46, 31),
 (47, 779),
 (83, 3),
 (84, 1),
 (92, 10),
 (96, 6),
 (105, 8),
 (107, 4),
 (118, 11),
 (120, 35),
 (140, 43),
 (152, 90),
 (153, 2),
 (158, 4),
 (166, 12),
 (167, 27),
 (177, 1),
 (181, 87),
 (185, 5),
 (200, 3),
 (214, 16),
 (224, 1),
 (227, 3),
 (228, 1),
 (232, 11),
 (233, 5),
 (234, 94),
 (244, 7),
 (247, 15),
 (256, 6),
 (263, 15),
 (273, 4)}

# ALL THE RESULTS ARE VOID. ERROR IN FC MAPS ----------------------------------------------------------------------------------------------------------------

# Results


1. GSR = 0, Fisher = 0   
Folder Name: functionalConnectivity_GSR0_fisher0  
{(ROI_number, Significant voxels)} at Q<0.5  :  
{(2, 1),
 (4, 2),
 (6, 1),
 (18, 16),
 (26, 8),
 (38, 11),
 (42, 4),
 (51, 28),
 (54, 1),
 (60, 33),
 (61, 1),
 (73, 1),
 (76, 27),
 (77, 8),
 (83, 6),
 (92, 4),
 (144, 1),
 (156, 22),
 (165, 37),
 (166, 85),
 (167, 283),
 (169, 3),
 (171, 1),
 (172, 4),
 (173, 12),
 (179, 6),
 (201, 1),
 (213, 1),
 (227, 1),
 (232, 7),
 (236, 226),
 (241, 2)}
 
2. GSR = 0, Fisher = 1
3. GSR = 1, Fisher = 0
4. GSR = 1, FIsher = 1